### ISSUE TITLE
Why not call the hash helper from the app?

### ISSUE THREAD

**Issue by pke** [andreszs/cordova-plugin-sms-retriever#16]
What's the reason Google advises against getting this hash from the running app? What's the risks here, the threat vector? @andreszs any idea?

**Comment by andreszs** [comment]
Honestly, I don't know, because the documentation does not explain why either. Probably due to some advanced security paranoia by the Android developers for very isolated cases on seriously compromised devices.

Anyway, the hash won't change as long as the app's signing keys don't change.

**Comment by pke** [comment]
Yeah I guess so too. Some weird exotic edge case. 

Thing is we have our app for alpha, beta, dev, and prod as 4 separate apps registered and I'd like to have as less dev-ops as possible in the deploy process. 
So yes, the hash needs to be calculated only once, but in case it does change (compromised signing key, some bug at googles signing tools) then somebody in the future needs to understand how to regenerate the keys (which is not a really easy flow).

Sending the hash with the request for the code to the server seems the best future proof solution to me at the moment. 
I'll document the process of manually generating the hash though (or maybe fastlane has some plugin to generate the hash).

**Comment by andreszs** [comment]
That's a good idea, I wish someone has documented something before I started working here too. Some brief comments on certain PHP files is all I got.

Regarding signatures, if for whatever reason it's changed, then users with your app won't be able to update it anymore, neither **Play Store** nor the [in-app-updates](https://github.com/andreszs/cordova-plugin-in-app-updates) plugin could update an APK with a signature mismatch. In fact not even adb itself is allowed to do so, for security reasons. Changing the signature practically means distributing a brand new app.
